Form/DataTransformer/ProviderDataTransformer.php

```diff
@@ -28,13 +29,19 @@ class ProviderDataTransformer implements DataTransformerInterface
     protected $options;
 
     /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
      * @param Pool   $pool
      * @param string $class
      * @param array  $options
      */
-    public function __construct(Pool $pool, $class, array $options = array())
+    public function __construct(Pool $pool, $class, array $options = array(), LoggerInterface $logger = null)
```

Comment:
Using LoggerAwareTrait could definitely be another option...

I've had a few alternatives offered for this issue...
I will add a global PR comment later today ([EDT](https://www.timeanddate.com/time/zones/edt)), summarize them, and wait for a definitive answer. I'd like to find a solution satisfying everyone.